sqlserver/datadog_checks/sqlserver/connection.py

```diff
@@ -28,6 +28,81 @@ class SQLConnectionError(Exception):
     pass
 
 
+# we're not including space in the set of invalid characters because it's a valid key character for adodbapi
+# best be more permissive here so that the driver and the database have the final say as to whether it's valid or not
+CONNECTION_STRING_SPECIAL_CHARACTERS = set('=;[]{}"\'')
```

Comment:
I don't know, I just included those because they were mentioned in the docs. But now that I think about it, best be even more permissive and only include the bare minimum characters here required for parsing. 
Updated: https://github.com/DataDog/integrations-core/pull/11664/commits/df340452db8f69c51c5edebfe639a8654526ab0c